### PR TITLE
[azurerm_role_assignment] Add support for Management Groups

### PR DESCRIPTION
**Pull request by nexxai**
Solves #4040

### CODE REVIEW

**Comment by nexxai** [comment]
I'm not entirely sure why this is failing, since the error is:

`cannot use props.Permissions (variable of type *[]authorization.Permission) as *[]authorization.Permission value in argument to flattenRoleDefinitionPermissions (typecheck)`

No matter how many times I read that, it seems like it's basically saying it failed because 1 != 1.

If anyone has any insight here, it would be greatly appreciated.

File: azurerm/resource_arm_role_assignment_test.go [hunk 1]
```diff
@@ -435,3 +460,22 @@ resource "azurerm_role_assignment" "test" {
...
+data "azurerm_subscription" "primary" {}
+
+data "azurerm_client_config" "test" {}
+
+data "azurerm_builtin_role_definition" "test" {
+	name = "Monitoring Reader"
+}
+
+data "azurerm_management_group" "test" {}
+	resource "azurerm_role_assignment" "test" {
```
Comment (katbyte) [inline azurerm/resource_arm_role_assignment_test.go]: Coould we fix this formatting:
```suggestion

resource "azurerm_role_assignment" "test" {
```

Comment (nexxai) [reply azurerm/resource_arm_role_assignment_test.go]: Done.

**Comment by katbyte** [comment]
It also looks like the test is failing for us with a:
```
Error: errors during apply: Error Deleting Azure AD Group with ID "bdb9f413-d452-4b3b-94c9-9c40c7c56255": graphrbac.GroupsClient#Delete: Failure responding to request: StatusCode=403 -- Original Error: autorest/azure: Service returned an error. Status=403 Code="Unknown" Message="Unknown service error" Details=[{"odata.error":{"code":"Authorization_RequestDenied","date":"2019-08-12T20:30:48","message":{"lang":"en","value":"Insufficient privileges to complete the operation."},"requestId":"d62d55df-f536-4147-b758-971160e6f19e"}}]
```

Are there any permissions we need to grant for this test ot pass?

**Comment by katbyte** [comment]
Sorry, that was the wrong test failure i copied, the correct one:
```
- This Data Source has been deprecated in favour of the 'azurerm_role_definition' resource that now can look up role definitions by names.
                
                As such this Data Source will be removed in v2.0 of the AzureRM Provider.
                
                - Missing required argument: The argument "group_id" is required, but no definition was found.
```

**Comment by nexxai** [comment]
It's still failing with this error: 

```    --- FAIL: TestAccAzureRMRoleAssignment/management (61.85s)
        --- FAIL: TestAccAzureRMRoleAssignment/management/assign (61.85s)
            testing.go:568: Step 0 error: After applying this step, the plan was not empty:
```

I haven't made any changes to any resources, so I don't know where else to look.  Can anyone take a peek and see if I'm missing something obvious?

**Comment by nexxai** [comment]
@katbyte The test is passing now.  Thanks for pointing that out.

**Pull request merged**